lib/linter/linter.js

```diff
@@ -1090,7 +1129,14 @@ function runRules(sourceCode, configuredRules, ruleMapper, parserName, languageO
             )
         );
 
-        const ruleListeners = timing.enabled ? timing.time(ruleId, createRuleListeners)(rule, ruleContext) : createRuleListeners(rule, ruleContext);
+        const ruleListenersReturn = (timing.enabled || stats)
+            ? timing.time(ruleId, createRuleListeners, stats)(rule, ruleContext) : createRuleListeners(rule, ruleContext);
+
+        const ruleListeners = stats ? ruleListenersReturn.result : ruleListenersReturn;
+
+        if (stats) {
+            storeTime(ruleListenersReturn.tdiff, { filename, type: "rules", key: ruleId }, slots);
```

Comment:
```suggestion
            storeTime(ruleListenersReturn.tdiff, { type: "rules", key: ruleId }, slots);
```

There's no need to pass `filename` anymore?